### ISSUE TITLE
Full node cannot finalize to the latest block

### ISSUE THREAD

**Issue by mmyyrroonn** [crustio/crust#785]
# Substrate Version: 3.0.0 release version
# Chain Type: Standalone
# Background
Our chain's finalized block got stuck at 2080276 five days ago. On that day, we sent several grandpa.noteStalled extrins to recover it. The delay is 1. It recovered once at around 2080200 but got stuck and again and recovered again at around 2080400. After the recovery, about 70% of our nodes are good to run. By copying the existing db to the other not-working nodes, most of them get back to normal.
# Problem
## 2021.12.07 10:30 update
For the new node syncing from scratch, the finalized block cannot pass 2080256. We enable several logs(afg, sync and some personally added logs) to observe the code logic and found two different situations
# 1. Some blocks(2080307, 2080313, 2080322) need justification and but cannot find justification in the block.
The code logic would go this path. https://github.com/paritytech/substrate/blob/264da3bedca973f0028c1e521e96327e2e72c35d/client/finality-grandpa/src/import.rs#L673-L681
```
None =>
if needs_justification {
	debug!(
		target: "afg",
		"Imported unjustified block #{} that enacts authority set change, waiting for finality for enactment.",
		number,
	);


	imported_aux.needs_justification = true;
},
```
```
2021-12-06 23:22:51.613  DEBUG tokio-runtime-worker afg: needs_justification is #true
2021-12-06 23:22:51.613  DEBUG tokio-runtime-worker afg: applied_changes none
2021-12-06 23:22:51.613  DEBUG tokio-runtime-worker afg: block 2080307 justification none
2021-12-06 23:22:51.613  DEBUG tokio-runtime-worker afg: Imported unjustified block #2080307 that enacts authority set change, waiting for finality for enactment.
```
After some time, it would produce the following logs.
```
 DEBUG tokio-runtime-worker afg: rough-view-1027: Starting new voter with set ID 604
2021-12-06 23:27:42.642  TRACE tokio-runtime-worker sync: Starting import of 128 blocks  (2080400..2080527)
2021-12-06 23:27:42.642  TRACE tokio-runtime-worker sync: Header 0x33bc…fc88 has 2 logs
2021-12-06 23:27:42.646  TRACE tokio-runtime-worker sync: Block imported but requires justification 2080307: 0x011a64a2715a5c19999609df8cff23685bec2dae91cd89b41e095cd2a1972523
2021-12-06 23:27:42.646  TRACE tokio-runtime-worker sync: Block imported but requires justification 2080313: 0xe329cc1375859249794a1404ee4c145b4c275189eb9c67dde8459895a498b3e8
2021-12-06 23:27:42.647  TRACE tokio-runtime-worker sync: Block imported but requires justification 2080322: 0x82731e572423959f05890bdeffd327f3fe80df70ead4c1920582b9e61d411c33
2021-12-06 23:27:42.647  TRACE tokio-runtime-worker sync: Block imported but requires justification 2080325: 0x1936118b373c7330ce0990e23c8f04fc421b92de7d8f6cba7221845c94f9992b
2021-12-06 23:27:42.647  TRACE tokio-runtime-worker sync: Block imported but requires justification 2080338: 0x2c68265c854f475e2c497e33966145eacd9cc29bbfe4e19d5a7b717fbeb64c9f
2021-12-06 23:27:42.647  TRACE tokio-runtime-worker sync: Block imported but requires justification 2080345: 0xa9c8db6929eb9233eebe46a062f424327a2fb269ba4d2c8a2ab815943fe9d26b
```

# 2. After our network recovery, some living nodes still got stuck at some blocks(2081517, 2084864, 2093568).
In these blocks, the following codes would be executed and we can see the following logs.

https://github.com/paritytech/substrate/blob/264da3bedca973f0028c1e521e96327e2e72c35d/client/finality-grandpa/src/import.rs#L655-L671
```
Some(justification) => {
let import_res = self.import_justification(
	hash,
	number,
	(GRANDPA_ENGINE_ID, justification),
	needs_justification,
	initial_sync,
);


import_res.unwrap_or_else(|err| {
	if needs_justification {
		debug!(target: "afg", "Imported block #{} that enacts authority set change with \
			invalid justification: {:?}, requesting justification from peers.", number, err);
		imported_aux.bad_justification = true;
		imported_aux.needs_justification = true;
	}
});
```

```
2021-12-06 23:36:09.356  DEBUG tokio-runtime-worker afg: needs_justification is #true
2021-12-06 23:36:09.356  DEBUG tokio-runtime-worker afg: applied_changes none
2021-12-06 23:36:09.356  DEBUG tokio-runtime-worker afg: block 2081517 justification some
2021-12-06 23:36:09.356  DEBUG tokio-runtime-worker afg: block 2081517 start verify
2021-12-06 23:36:09.359  DEBUG tokio-runtime-worker afg: Bad signature on message from Public(0137b2528f97dd5bab81d2a18a7584faf9e1a8c4c37c3fcc30f4db5c4f3e9201 (5C6JT5Ln...))
2021-12-06 23:36:09.359  DEBUG tokio-runtime-worker afg: decode failed
2021-12-06 23:36:09.361  DEBUG tokio-runtime-worker afg: Imported block #2081517 that enacts authority set change with invalid justification: ClientImport("bad justification for header: invalid signature for precommit in grandpa justification"), requesting justification from peers.
```
or
```
 make authority change is none
2021-12-06 23:47:39.080  DEBUG tokio-runtime-worker afg: needs_justification is #false
2021-12-06 23:47:39.080  DEBUG tokio-runtime-worker afg: applied_changes none
2021-12-06 23:47:39.080  DEBUG tokio-runtime-worker afg: block 2084864 justification some
2021-12-06 23:47:39.090  DEBUG tokio-runtime-worker afg: block 2084864 start verify
2021-12-06 23:47:39.090  DEBUG tokio-runtime-worker afg: decode failed
2021-12-06 23:47:39.091  TRACE tokio-runtime-worker sync: Block imported successfully Some(2084864) (0x49c1…1d09)
```
I'm not sure whether it would recover after problem 1 is solved.

The best block can continue growing until the latest best block. However, there would be lots of the following logs to request justification for blocks but got an empty responses.
```
2021-12-07 08:59:29.465  TRACE tokio-runtime-worker sync: Empty justification response from PeerId("12D3KooWAknKKJXMPw42JHDRTRHnxEzSFUeJFEgWDF4w8tde8U2L") for (0x9a44b224f08717c5ca1a03d2d70fd58ecb46af0df415f50c006b6635cf345241, 2080298)
2021-12-07 08:59:29.723  TRACE tokio-runtime-worker sync: Sending justification request to PeerId("12D3KooWACoCdQ9d14dW42TdoLTBik9L2C8YVbkSE1w841wKz2Xc") for (0xd8149b52712ae9d934cdbac7a81015c516f88a496024111ec42a6f4b27607c66, 2080299)
2021-12-07 08:59:29.723  TRACE tokio-runtime-worker sync: Sending justification request to PeerId("12D3KooWAknKKJXMPw42JHDRTRHnxEzSFUeJFEgWDF4w8tde8U2L") for (0x7e92911cfafff66a7fc51046281a06be73771a0159299d45d568642cb0ea0106, 2080293)
2021-12-07 08:59:29.754  TRACE tokio-runtime-worker sync: BlockResponse 0 from 12D3KooWAknKKJXMPw42JHDRTRHnxEzSFUeJFEgWDF4w8tde8U2L with 0 blocks
2021-12-07 08:59:29.754  TRACE tokio-runtime-worker sync: Peer PeerId("12D3KooWAknKKJXMPw42JHDRTRHnxEzSFUeJFEgWDF4w8tde8U2L") provided empty response for justification request 0x7e92911cfafff66a7fc51046281a06be73771a0159299d45d568642cb0ea0106
2021-12-07 08:59:29.754  TRACE tokio-runtime-worker sync: Empty justification response from PeerId("12D3KooWAknKKJXMPw42JHDRTRHnxEzSFUeJFEgWDF4w8tde8U2L") for (0x7e92911cfafff66a7fc51046281a06be73771a0159299d45d568642cb0ea0106, 2080293)
2021-12-07 08:59:29.776  TRACE tokio-runtime-worker sync: BlockResponse 0 from 12D3KooWACoCdQ9d14dW42TdoLTBik9L2C8YVbkSE1w841wKz2Xc with 1 blocks
2021-12-07 08:59:29.777  TRACE tokio-runtime-worker sync: Empty justification response from PeerId("12D3KooWACoCdQ9d14dW42TdoLTBik9L2C8YVbkSE1w841wKz2Xc") for (0xd8149b52712ae9d934cdbac7a81015c516f88a496024111ec42a6f4b27607c66, 2080299)
2021-12-07 08:59:29.858  TRACE tokio-runtime-worker sync: BlockResponse 0 from 12D3KooWGes8bR1BBe9fmfZyX3MNhJRJQXyJDrGNcTfiwSMYpRxS with 1 blocks
2021-12-07 08:59:29.858  TRACE tokio-runtime-worker sync: Empty justification response from PeerId("12D3KooWGes8bR1BBe9fmfZyX3MNhJRJQXyJDrGNcTfiwSMYpRxS") for (0x2fb651221ba328523210189b5cc710ee2ba7a97fbd58754cc166962a9958cb3d, 2081517)
2021-12-07 08:59:29.948  TRACE tokio-runtime-worker sync: Sending justification request to PeerId("12D3KooWKWcQksGCN8YrQuZTbueyJtUMg7SBSkT3sqSn2HqAhXAP") for (0xd8149b52712ae9d934cdbac7a81015c516f88a496024111ec42a6f4b27607c66, 2080299)
2021-12-07 08:59:29.948  TRACE tokio-runtime-worker sync: Sending justification request to PeerId("12D3KooWACoCdQ9d14dW42TdoLTBik9L2C8YVbkSE1w841wKz2Xc") for (0xf6266ef3b26af181139c60aae3aa7e9ca1048aa1cc020df35fbf761183798377, 2080304)
2021-12-07 08:59:29.983  TRACE tokio-runtime-worker afg: Polling round 1, state = State { prevote_ghost: None, finalized: None, estimate: None, completable: false }, step = Some(Prevoted)
2021-12-07 08:59:29.983  TRACE tokio-runtime-worker sync: Sending justification request to PeerId("12D3KooWGes8bR1BBe9fmfZyX3MNhJRJQXyJDrGNcTfiwSMYpRxS") for (0x9d1054a55d8b567c2e3d9d54131446aa7cb4963399d758067a583f725b49002b, 2080296)
2021-12-07 08:59:30.005  TRACE tokio-runtime-worker sync: BlockResponse 0 from 12D3KooWACoCdQ9d14dW42TdoLTBik9L2C8YVbkSE1w841wKz2Xc with 1 blocks
2021-12-07 08:59:30.005  TRACE tokio-runtime-worker sync: Empty justification response from PeerId("12D3KooWACoCdQ9d14dW42TdoLTBik9L2C8YVbkSE1w841wKz2Xc") for (0xf6266ef3b26af181139c60aae3aa7e9ca1048aa1cc020df35fbf761183798377, 2080304)
2021-12-07 08:59:30.072  TRACE tokio-runtime-worker sync: BlockResponse 0 from 12D3KooWKWcQksGCN8YrQuZTbueyJtUMg7SBSkT3sqSn2HqAhXAP with 1 blocks
2021-12-07 08:59:30.073  TRACE tokio-runtime-worker sync: Empty justification response from PeerId("12D3KooWKWcQksGCN8YrQuZTbueyJtUMg7SBSkT3sqSn2HqAhXAP") for (0xd8149b52712ae9d934cdbac7a81015c516f88a496024111ec42a6f4b27607c66, 2080299)
2021-12-07 08:59:30.073  TRACE tokio-runtime-worker sync: BlockResponse 0 from 12D3KooWGes8bR1BBe9fmfZyX3MNhJRJQXyJDrGNcTfiwSMYpRxS with 1 blocks
2021-12-07 08:59:30.073  TRACE tokio-runtime-worker sync: Empty justification response from PeerId("12D3KooWGes8bR1BBe9fmfZyX3MNhJRJQXyJDrGNcTfiwSMYpRxS") for (0x9d1054a55d8b567c2e3d9d54131446aa7cb4963399d758067a583f725b49002b, 2080296)
2021-12-07 08:59:30.269  TRACE tokio-runtime-worker sync: Sending justification request to PeerId("12D3KooWAknKKJXMPw42JHDRTRHnxEzSFUeJFEgWDF4w8tde8U2L") for (0x9d1054a55d8b567c2e3d9d54131446aa7cb4963399d758067a583f725b49002b, 2080296)
2021-12-07 08:59:30.325  TRACE tokio-runtime-worker sync: BlockResponse 0 from 12D3KooWAknKKJXMPw42JHDRTRHnxEzSFUeJFEgWDF4w8tde8U2L with 1 blocks
2021-12-07 08:59:30.325  TRACE tokio-runtime-worker sync: Empty justification response from PeerId("12D3KooWAknKKJXMPw42JHDRTRHnxEzSFUeJFEgWDF4w8tde8U2L") for (0x9d1054a55d8b567c2e3d9d54131446aa7cb4963399d758067a583f725b49002b, 2080296)
2021-12-07 08:59:30.325  TRACE tokio-runtime-worker sync: Sending justification request to PeerId("12D3KooWAknKKJXMPw42JHDRTRHnxEzSFUeJFEgWDF4w8tde8U2L") for (0xf6266ef3b26af181139c60aae3aa7e9ca1048aa1cc020df35fbf761183798377, 2080304)
```

**Comment by mmyyrroonn** [comment]
## 2021.12.07 11:30 update
Resync the node again would get different logs. The block become 2080308
```
 👶 New epoch 3503 launching at block 0xe9a7…16ba (block slot 273071325 >= start slot 273071305).
2021-12-07 07:06:35.874   INFO tokio-runtime-worker babe: 👶 Next epoch starts at slot 273071905
2021-12-07 07:06:35.875  DEBUG tokio-runtime-worker afg: here number is #2080308
2021-12-07 07:06:35.875  DEBUG tokio-runtime-worker afg: Inserting potential standard set change signaled at block (2080308, 0xe9a707867058763a5425f5e2c3c2dfc6de7a15bce9e50784cf13021b88fe16ba) (delayed by 0 blocks).
2021-12-07 07:06:35.875  DEBUG tokio-runtime-worker afg: There are now 7 alternatives for the next pending standard change (roots), and a total of 25 pending standard changes (across all forks).
2021-12-07 07:06:35.875  DEBUG tokio-runtime-worker afg: make authority change is standard
2021-12-07 07:06:35.877   INFO tokio-runtime-worker substrate: ✨ Imported #2080308 (0xe9a7…16ba)
2021-12-07 07:06:35.878  TRACE tokio-runtime-worker afg: Polling round 1, state = State { prevote_ghost: None, finalized: None, estimate: None, completable: false }, step = Some(Prevoted)
2021-12-07 07:06:35.878  DEBUG tokio-runtime-worker afg: needs_justification is #true
2021-12-07 07:06:35.878  DEBUG tokio-runtime-worker afg: applied_changes standard
2021-12-07 07:06:35.878  DEBUG tokio-runtime-worker afg: block 2080308 justification none
2021-12-07 07:06:35.878  TRACE tokio-runtime-worker afg: Polling round 1, state = State { prevote_ghost: None, finalized: None, estimate: None, completable: false }, step = Some(Prevoted)
2021-12-07 07:06:35.878  DEBUG tokio-runtime-worker afg: Imported unjustified block #2080308 that enacts authority set change, waiting for finality for enactment.
```

**Comment by mmyyrroonn** [comment]
## 2021.12.07 14:00 update
Resync the node again without bootnodes, in and out peers are set to 0 and with some reserved nodes. We collected three kinds of logs now.
### 1
```
2021-12-07 07:06:35.848  TRACE tokio-runtime-worker sync: Announcing block 0xa55fa44e20f86a55aca39d258d687fcd3e602258090dbc29b0b96d34303db35a to 12D3KooWG9LioBPrRhdDVBGLDBT5e7iA4niuQ9ip7pASAx1mbZHj
2021-12-07 07:06:35.849  DEBUG tokio-runtime-worker afg: here number is #2080307
2021-12-07 07:06:35.849  DEBUG tokio-runtime-worker afg: make authority change is none
2021-12-07 07:06:35.850  DEBUG tokio-runtime-worker afg: needs_justification is #false
2021-12-07 07:06:35.850  DEBUG tokio-runtime-worker afg: applied_changes none
2021-12-07 07:06:35.850  DEBUG tokio-runtime-worker afg: block 2080307 justification none
2021-12-07 07:06:35.850  TRACE tokio-runtime-worker sync: Block imported successfully Some(2080307) (0x9bca…1042)
2021-12-07 07:06:35.850  TRACE tokio-runtime-worker afg: Polling round 1, state = State { prevote_ghost: None, finalized: None, estimate: None, completable: false }, step = Some(Prevoted)
2021-12-07 07:06:35.850   INFO tokio-runtime-worker substrate: ✨ Imported #2080307 (0x9bca…1042)
2021-12-07 07:06:35.850  TRACE tokio-runtime-worker sync: Header 0xe9a7…16ba has 162 logs
```

### 2
```
r sync: Announcing block 0xa55fa44e20f86a55aca39d258d687fcd3e602258090dbc29b0b96d34303db35a to 12D3KooWG9LioBPrRhdDVBGLDBT5e7iA4niuQ9ip7pASAx1mbZHj
2021-12-07 07:06:35.849  DEBUG tokio-runtime-worker afg: here number is #2080307
2021-12-07 07:06:35.849  DEBUG tokio-runtime-worker afg: make authority change is none
2021-12-07 07:06:35.850  DEBUG tokio-runtime-worker afg: needs_justification is #false
2021-12-07 07:06:35.850  DEBUG tokio-runtime-worker afg: applied_changes none
2021-12-07 07:06:35.850  DEBUG tokio-runtime-worker afg: block 2080307 justification none
2021-12-07 07:06:35.850  TRACE tokio-runtime-worker sync: Block imported successfully Some(2080307) (0x9bca…1042)
2021-12-07 07:06:35.850  TRACE tokio-runtime-worker afg: Polling round 1, state = State { prevote_ghost: None, finalized: None, estimate: None, completable: false }, step = Some(Prevoted)
2021-12-07 07:06:35.850   INFO tokio-runtime-worker substrate: ✨ Imported #2080307 (0x9bca…1042)
2021-12-07 07:06:35.850  TRACE tokio-runtime-worker sync: Header 0xe9a7…16ba has 162 logs
2021-12-07 07:06:35.850  DEBUG tokio-runtime-worker sync: Reannouncing block 0x9bcaefc05db9109f4b89fc2abffe3050562d6775a433c9442014f85916bd1042 is_best: false
2021-12-07 07:06:35.869  TRACE tokio-runtime-worker sync: Announcing block 0x9bcaefc05db9109f4b89fc2abffe3050562d6775a433c9442014f85916bd1042 to 12D3KooWRbHX2Kq4i43R1i5DeLgeY5zFqdc3NUrykiWnMdork7SP
2021-12-07 07:06:35.869  TRACE tokio-runtime-worker sync: Announcing block 0x9bcaefc05db9109f4b89fc2abffe3050562d6775a433c9442014f85916bd1042 to 12D3KooWACoCdQ9d14dW42TdoLTBik9L2C8YVbkSE1w841wKz2Xc
2021-12-07 07:06:35.869  TRACE tokio-runtime-worker sync: Announcing block 0x9bcaefc05db9109f4b89fc2abffe3050562d6775a433c9442014f85916bd1042 to 12D3KooWGes8bR1BBe9fmfZyX3MNhJRJQXyJDrGNcTfiwSMYpRxS
2021-12-07 07:06:35.869  TRACE tokio-runtime-worker sync: Announcing block 0x9bcaefc05db9109f4b89fc2abffe3050562d6775a433c9442014f85916bd1042 to 12D3KooWEPKCABvLEtLZzhpb3SXQjNwo7P4hMrq2NcBfF7oLb2gD
2021-12-07 07:06:35.873  TRACE tokio-runtime-worker sync: Announcing block 0x9bcaefc05db9109f4b89fc2abffe3050562d6775a433c9442014f85916bd1042 to 12D3KooWR95hfCZjSeGiFTaEBdKheSsWVLfsi57WBQ8qFbxbkR3U
2021-12-07 07:06:35.873  TRACE tokio-runtime-worker sync: Announcing block 0x9bcaefc05db9109f4b89fc2abffe3050562d6775a433c9442014f85916bd1042 to 12D3KooWFeEPrNxL5FdU7GZxk8WTszYj9o9fzAeCHxqDXGpXV2Qq
2021-12-07 07:06:35.873  TRACE tokio-runtime-worker sync: Announcing block 0x9bcaefc05db9109f4b89fc2abffe3050562d6775a433c9442014f85916bd1042 to 12D3KooWHKGT3as4AcQPqjkSP3LG9JWASMA5Z77AUsc4qj9S93mx
2021-12-07 07:06:35.873  TRACE tokio-runtime-worker sync: Announcing block 0x9bcaefc05db9109f4b89fc2abffe3050562d6775a433c9442014f85916bd1042 to 12D3KooWDduS45zvrCDeEbgpVrJC9zZsqPS4U9S2eemC4a56a2ya
2021-12-07 07:06:35.873  TRACE tokio-runtime-worker sync: Announcing block 0x9bcaefc05db9109f4b89fc2abffe3050562d6775a433c9442014f85916bd1042 to 12D3KooWAknKKJXMPw42JHDRTRHnxEzSFUeJFEgWDF4w8tde8U2L
2021-12-07 07:06:35.873  TRACE tokio-runtime-worker sync: Announcing block 0x9bcaefc05db9109f4b89fc2abffe3050562d6775a433c9442014f85916bd1042 to 12D3KooWKWcQksGCN8YrQuZTbueyJtUMg7SBSkT3sqSn2HqAhXAP
2021-12-07 07:06:35.873  TRACE tokio-runtime-worker sync: Announcing block 0x9bcaefc05db9109f4b89fc2abffe3050562d6775a433c9442014f85916bd1042 to 12D3KooWG9LioBPrRhdDVBGLDBT5e7iA4niuQ9ip7pASAx1mbZHj
2021-12-07 07:06:35.874   INFO tokio-runtime-worker babe: 👶 New epoch 3503 launching at block 0xe9a7…16ba (block slot 273071325 >= start slot 273071305).
2021-12-07 07:06:35.874   INFO tokio-runtime-worker babe: 👶 Next epoch starts at slot 273071905
2021-12-07 07:06:35.875  DEBUG tokio-runtime-worker afg: here number is #2080308
2021-12-07 07:06:35.875  DEBUG tokio-runtime-worker afg: Inserting potential standard set change signaled at block (2080308, 0xe9a707867058763a5425f5e2c3c2dfc6de7a15bce9e50784cf13021b88fe16ba) (delayed by 0 blocks).
2021-12-07 07:06:35.875  DEBUG tokio-runtime-worker afg: There are now 7 alternatives for the next pending standard change (roots), and a total of 25 pending standard changes (across all forks).
2021-12-07 07:06:35.875  DEBUG tokio-runtime-worker afg: make authority change is standard
2021-12-07 07:06:35.877   INFO tokio-runtime-worker substrate: ✨ Imported #2080308 (0xe9a7…16ba)
```

### 3
```
2021-12-07 14:04:43.176  TRACE tokio-runtime-worker sync: Too many blocks in the queue.
2021-12-07 14:04:43.475  DEBUG tokio-runtime-worker afg: needs_justification is #true
2021-12-07 14:04:43.475  DEBUG tokio-runtime-worker afg: applied_changes none
2021-12-07 14:04:43.475  DEBUG tokio-runtime-worker afg: block 2080307 justification none
2021-12-07 14:04:43.475  DEBUG tokio-runtime-worker afg: Imported unjustified block #2080307 that enacts authority set change, waiting for finality for enactment.
2021-12-07 14:04:43.475  TRACE tokio-runtime-worker sync: Block imported successfully Some(2080307) (0x011a…2523)

2021-12-07 14:05:28.339  TRACE tokio-runtime-worker sync: Too many blocks in the queue.
2021-12-07 14:05:28.339  TRACE tokio-runtime-worker sync: Too many blocks in the queue.
2021-12-07 14:05:28.619  DEBUG tokio-runtime-worker afg: needs_justification is #false
2021-12-07 14:05:28.619  DEBUG tokio-runtime-worker afg: applied_changes none
2021-12-07 14:05:28.619  DEBUG tokio-runtime-worker afg: block 2080308 justification none
2021-12-07 14:05:28.619  TRACE tokio-runtime-worker sync: Block imported successfully Some(2080308) (0x6c54…b669)
2021-12-07 14:05:28.619  TRACE tokio-runtime-worker sync: Header 0x0d38…8a81 has 2 logs
```

**Comment by andresilva** [comment]
What was the reason in the first place for requiring to issue a forced change? You also mention that finality initially got stuck at block #2080276 but you issued a `grandpa.noteStalled` to recover at block #2080200, which seems wrong since you're moving finality backwards. Can you tell me exactly which `grandpa.noteStalled` extrinsics were successfully applied?

**Comment by badkk** [comment]
> What was the reason in the first place for requiring to issue a forced change? You also mention that finality initially got stuck at block #2080276 but you issued a `grandpa.noteStalled` to recover at block #2080200, which seems wrong since you're moving finality backwards. Can you tell me exactly which `grandpa.noteStalled` extrinsics were successfully applied?

## Reason

Most of validators(over 26, total 80) go offline, *block finality* stuck on `#2080276`

## Whole process

The whole process shows like below 😭 
 
![6121638883999_ pic_hd](https://user-images.githubusercontent.com/10335300/145038441-0acc7114-98ce-4a63-a367-5345acb12750.jpg)

**Comment by mmyyrroonn** [comment]
> What was the reason in the first place for requiring to issue a forced change? You also mention that finality initially got stuck at block #2080276 but you issued a `grandpa.noteStalled` to recover at block #2080200, which seems wrong since you're moving finality backwards. Can you tell me exactly which `grandpa.noteStalled` extrinsics were successfully applied?

Sorry. Made a mistake. Please see the above picture.

**Comment by badkk** [comment]
Fixed on #790
